app/helpers/msgTx.js

```diff
@@ -182,84 +186,136 @@ export function decodeRawTransaction(rawTx) {
   tx.version = version & 0xffff;
   tx.serType = version >> 16;
   position += 4;
-  let first = rawTx.readUInt8(position);
-  position += 1;
-  switch (first) {
-    case 0xfd:
-      tx.numInputs = rawTx.readUInt16LE(position);
-      position += 2;
-      break;
-    case 0xfe:
-      tx.numInputs = rawTx.readUInt32LE(position);
+
+  if (tx.serType != SERTYPE_ONLYWITNESS) {
```

Comment:
I think we should use a strict comparison here instead of loose one.

Reply:
Nice catch. Updated.